artemis-server/src/main/java/org/apache/activemq/artemis/core/paging/impl/PagingStoreImpl.java

```diff
@@ -189,13 +198,45 @@ public PagingStoreImpl(final SimpleString address,
       this.usingGlobalMaxSize = pagingManager.isUsingGlobalSize();
    }
 
+   private static SizeAwareMetric getGlobalMetric(PagingManager manager) {
+      if (manager instanceof PagingManagerImpl) {
+         return ((PagingManagerImpl) manager).getSizeAwareMetric();
+      } else {
+         // for unit / mock tests
+         SizeAwareMetric metric = new SizeAwareMetric(-1, -1, -1, -1, -1, 1, 1, 0, 0);
+         metric.addOnSizeCallback(manager::addSize, false);
+         metric.addOnSizeCallback((l) -> manager.addSize(l, true), true);
+         return metric;
+      }
+   }
```

Comment:
I did not want to add SizeAwareMetric to the interface... I wanted it to be an internal implementation detail.

if the implementation (PagingManagerImpl) is using it, it could use it on a parent call for the metric. Otherwise the proxy Metric would use the already existing API.

I'm adding it to the interface now with a default implementation returning the proxy. I didn't want to do it.. but I will do it just to settle this.

Reply:
actually, it will look better to just use the SizeAwareMetric.

Reply:
Having internal detail is fine. The main point was really the other bit though, its generally a bad idea for non-test code to be just suddenly returning objects only intended for [certain] tests. At some point something will unexpectedly get one of those instead of the real thing, when theres no reason they really should know the behaviour is there. Better for tests that need to to know and apply the test logic they need.

Reply:
the other bits is just returning a proxy to the pagingManager API.

anyway I'm adding SizeAwareMetric to the public api with a default implementation to the proxy version now.